src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -1084,6 +1125,286 @@ bool FullyConnected::useSparseWeightsDecompression() {
+
+                        if (quant.size() == 1) {
+                            std::fill(requant.get(), requant.get() + N, quant[0]);
+                        } else {
+                            memcpy(requant.get(), quant.data(), quant.size() * sizeof(float));
+                        }
+                        param.postops_type = static_cast<llmdnn::postops_types>(param.postops_type | llmdnn::QUANT);
+                    } else {
+                        valid = false;
+                        break;
+                    }
+                } else {
+                    valid = false;
+                    break;
+                }
+            }
+            return valid;
+        }
+    }
+
+    return false;
+}
+
+MemoryPtr FullyConnected::convertMemoryPrecision(MemoryPtr memPtr, const InferenceEngine::Precision prec) {
+    if (prec == memPtr->getDesc().getPrecision())
+        return memPtr;
+
+    auto newMemDescPtr = memPtr->getDescPtr()->cloneWithNewPrecision(prec);
+
+    Memory srcMemory{ getEngine(), memPtr->getDescWithType<DnnlMemoryDesc>(), memPtr->getData() };
+    MemoryPtr newMemPtr = std::make_shared<Memory>(getEngine(), newMemDescPtr);
+    node::Reorder::reorderData(srcMemory, *newMemPtr, context->getParamsCache());
+
+    return newMemPtr;
+}
+
+/*
+ Invoke the llmdnn FC kernel.
+ - In the very first execution round, the BF16 weight is needed to initialize the kernel.
+   Becides, the weight will be repacked and cached in kernel.
+ - In the other execution rounds, the weight is not needed at all since the it has been cached in the first run.
+   So, just passing the `nullptr` is OK.
+ */
+void FullyConnected::primExecLLMFc(void* weight) {
+    // src
+    auto srcPtr = getParentEdgeAt(DATA_ID)->getMemoryPtr();
+    auto* src = srcPtr->getData();
+    auto& data_dims = srcPtr->getStaticDims();
+
+    // dst
+    auto* dst = getChildEdgeAt(0)->getMemoryPtr()->getData();
+
+    // M, N, K
+    auto M = data_dims[0];
+    auto N = weightDims[0];
+    auto K = weightDims[1];
+    if (data_dims.size() == 3) {
+        M *= data_dims[1];
+        assert(K == data_dims[2]);
+    } else {
+        assert(K == data_dims[1]);
+    }
+    auto work_amount = rnd_up(N, 32) / 32;
+    float* bias = nullptr;
+    if (withBiases) {
+        bias = biasRnd.get();
+    }
+    auto thread_num_fcllm = fcLLMs.size();
+    size_t lda, ldb, ldc;
+    lda = K * dnnl::memory::data_type_size(inputDataType);
+    ldb = K * dnnl::memory::data_type_size(inputDataType);
+    ldc = N * dnnl::memory::data_type_size(outputDataType);
+    parallel_for(thread_num_fcllm, [&](size_t tid) {
+        size_t start {0}, end {0};
+        dnnl::impl::balance211(work_amount, thread_num_fcllm, tid, start, end);
+        size_t n0 = start * 32;
+        size_t n1 = std::min(end * 32, N);
+        if (n0 >= N) return;
+
+        llmdnn::fc_kernel_execute(fcLLMs[tid].get(), src, weight, dst, lda, ldb, ldc, M, N, K, n0, n1,
```

Comment:
Then leading dimension should follow common definition. And leading dimension is no stride. Please refer to mkl for example
<html>
<body>
<!--StartFragment--><dt class="dlterm" style="box-sizing: border-box; line-height: 1.42857; font-weight: 700; color: rgb(38, 38, 38); font-family: intel-clear, tahoma, Helvetica, helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="parmname" style="box-sizing: border-box; font-family: monospace;">lda</span></dt><dd style="box-sizing: border-box; line-height: 1.42857; margin-left: 0px; color: rgb(38, 38, 38); font-family: intel-clear, tahoma, Helvetica, helvetica, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><p style="box-sizing: border-box; margin: 0px 0px 11px; line-height: 1.25;">Specifies the leading dimension of<span> </span><var style="box-sizing: border-box;">a</var><span> </span>as declared in the calling (sub)program.</p>

  | transa=CblasNoTrans | transa=CblasTrans or transa=CblasConjTrans
-- | -- | --
Layout = CblasColMajor | lda must be at least max(1, m). | lda must be at least max(1, k)
Layout = CblasRowMajor | lda must be at least max(1, k) | lda must be at least max(1, m).

</dd><!--EndFragment-->
</body>
</html>

Reply:
Without stride the block scheme will not work, name with ldx is renamed to stride_x to avoid misunderstanding.